services/scabbard/libscabbard/src/service/v3/consensus/consensus_action_runner/notify_observer/command.rs

```diff
@@ -0,0 +1,201 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Contains an implementation of a `NotifyObserver` that operates soley on returning commands
+
+use std::sync::Arc;
+
+use rand::{distributions::Alphanumeric, Rng};
+
+use splinter::error::InternalError;
+use splinter::service::FullyQualifiedServiceId;
+use splinter::store::command::StoreCommand;
+
+use crate::service::v3::consensus::consensus_action_runner::commands::notifications::{
+    AddCommitEntryCommand, AddEventCommand, UpdateCommitEntryCommand,
+};
+use crate::store::two_phase::action::ConsensusActionNotification;
+use crate::store::{
+    commit::{CommitEntryBuilder, ConsensusDecision},
+    event::ScabbardConsensusEvent,
+    two_phase::event::Scabbard2pcEvent,
+};
+use crate::store::{ScabbardStore, ScabbardStoreFactory};
+
+use super::NotifyObserver;
+
+const RANDOM_VALUE_LENGTH: usize = 32;
+
+/// Implementation of `NotifyObserver` that operates on `StoreCommands`
+pub struct CommandNotifyObserver<C: 'static> {
+    store_factory: Arc<dyn ScabbardStoreFactory<C>>,
+    store: Box<dyn ScabbardStore>,
+}
+
+impl<C: 'static> CommandNotifyObserver<C> {
+    /// Create a new `CommandNotifyObserver`
+    ///
+    /// # Arguments
+    ///
+    /// * `store_factory` - The scabbard store factory to be used by commands
+    /// * `store` - The scabbard store used to fetch current state
+    pub fn new(
+        store_factory: Arc<dyn ScabbardStoreFactory<C>>,
+        store: Box<dyn ScabbardStore>,
+    ) -> Self {
+        Self {
+            store_factory,
+            store,
+        }
+    }
+}
+
+impl<C: 'static> NotifyObserver<C> for CommandNotifyObserver<C> {
+    /// Notify components about consensus notification
+    ///
+    /// # Arguments
+    ///
+    /// * `notification` - The notification that needs to be handled
+    /// * `service_id` - The service ID of of the service the notification is for
+    /// * `epoch` - The current epoch of the consensus algorithm
+    fn notify(
+        &self,
+        notification: ConsensusActionNotification,
+        service_id: &FullyQualifiedServiceId,
+        epoch: u64,
+    ) -> Result<Vec<Box<dyn StoreCommand<Context = C>>>, InternalError> {
+        let mut commands: Vec<Box<dyn StoreCommand<Context = C>>> = Vec::new();
+        match notification {
+            // Generates a new value to agree on and creates commands to add a commit entry to
+            // track the value and an new event to start agreement on that value
+            ConsensusActionNotification::RequestForStart() => {
+                let s: String = rand::thread_rng()
```

Comment:
Instead of adding a new dep for this temporary use - we could just use the string value of system time millis

Reply:
good idea, ill update that

Reply:
updated